### PR TITLE
dgsh-unstable: Init at 2017-02-05

### PR DESCRIPTION
**Pull request by vrthra**
Dgsh is the directed graph shell

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/shells/dgsh/default.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{ stdenv, fetchgit, autoconf, automake, pkgconfig,
+  libtool, check, bison, git, gperf,
+  perl, rsync, texinfo, help2man, gettext
+}:
+
+stdenv.mkDerivation rec {
+  name = "dgsh-${version}";
+  version = "nightly-2017-02-05";
```
Comment (rycee) [inline pkgs/shells/dgsh/default.nix]: Versions can't start with a letter. This should be
```
name = "dgsh-unstable";
version = "2017-02-05";
```
See description in the [package naming](https://nixos.org/nixpkgs/manual/#sec-package-naming) section of the Nixpkgs manual.

Comment (vrthra) [inline pkgs/shells/dgsh/default.nix]: Thanks, I have updated the name and version

File: pkgs/shells/dgsh/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+  src = fetchFromGitHub {
+    owner = "dspinellis";
+    repo = "dgsh";
+    rev = "bc4fc2e8009c069ee4df5140c32a2fc15d0acdec";
+    sha256 = "0k3hmnarz56wphw45mabn5zcc427l5p77jldh1qqy89pxqy1wnql";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [ autoconf automake pkgconfig libtool check
```
Comment (joachifm) [inline pkgs/shells/dgsh/default.nix]: I wonder what `rsync` is used for during the build?

Comment (vrthra) [inline pkgs/shells/dgsh/default.nix]: It isn't used though its existence is checked. Updated without rsync.

**Pull request merged**